### Scope

In [32]:
from datamodel import RegionExtractor
from labeledarray import LabeledArrayConstructor, LabeledArray
from os.path import join
import numpy as np
import matplotlib.pyplot as plt
from datamodel import Field, ImageDict
from collections import OrderedDict
%matplotlib inline

Create a labeled array for a single position.

In [20]:
folders = []
for al in ('A', 'B', 'C','D','E', 'F', 'G'):
    for s in (4, 5, 6, 7):
        for s2 in (0, 1, 2, 3):
            folders.append('{0}{1}_Site_{2}'.format(al, s, s2))
folders.remove('C7_Site_2')

In [3]:
imdict = ImageDict()
imdict.load(join('/Volumes/covert-lab/share/Taka/2017/eTK078/images/', folders[0]+'.tif'))
rext = RegionExtractor(imdict, objnames=['nuc'])
reg = rext._extract()

In [15]:
darr = LabeledArrayConstructor(reg).construct()
print darr.shape
print darr.labelarr[:5]

(104, 302)
[['nuc' 'area' '']
 ['nuc' 'centroid0' '']
 ['nuc' 'centroid1' '']
 ['nuc' 'eccentricity' '']
 ['nuc' 'label' '']]


Labeled array stores all the values extratced from segmentation. In this case it has 104 properties and 302 cells.  
Each rows corresponds to each rows of labelarr which indicates properties. You can extract data by indexing through properties you want.

In [16]:
print darr['nuc', 'AMCA', 'median_intensity'].shape

(2, 302)


In this case, it shows the median intensity from nuclear segmentation in AMCA channel.  
There are two rows since there are values from preif and postif.

In [19]:
darr.save(join('/Volumes/covert-lab/share/Taka/2017/eTK078/values/', folders[0]))
la = LabeledArray()
la = la.load(join('/Volumes/covert-lab/share/Taka/2017/eTK078/values/', folders[0]+'.npz'))
print la['nuc', 'AMCA', 'median_intensity'].shape

(2, 302)


It still preserves labels and values as it was even you save and load.

In [55]:
# Iterate over folders
for folder in folders:
    imdict = ImageDict()
    imdict.load(join('/Volumes/covert-lab/share/Taka/2017/eTK078/images/', folder+'.tif'))
    rext = RegionExtractor(imdict, objnames=['nuc'])
    reg = rext._extract()
    darr = LabeledArrayConstructor(reg).construct()
    darr.save(join('/Volumes/covert-lab/share/Taka/2017/eTK078/values/', folder))

Now start plotting the stats just for the E2 staining. It is Farred channel.

In [56]:
# notags for HeLa WT, tags for HeLa expressing epitopes.
notags_folders = [i for i in folders if '4_Site' in i]
tags_folders = [i for i in folders if '5_Site' in i]

In [59]:
notags, tags = OrderedDict(), OrderedDict()
for folder in notags_folders:
    la = LabeledArray()
    notags[i] = la.load(join('/Volumes/covert-lab/share/Taka/2017/eTK078/values/', folder+'.npz'))
for folder in tags_folders:
    la = LabeledArray()
    tags[i] = la.load(join('/Volumes/covert-lab/share/Taka/2017/eTK078/values/', folder+'.npz'))